bitcoin/src/blockdata/witness.rs

```diff
@@ -139,22 +139,19 @@ crate::internal_macros::define_extension_trait! {
             self.push(signature.serialize())
         }
 
-        /// Get Tapscript following BIP341 rules regarding accounting for an annex.
+        /// Returns the leaf script with its version but without the merkle proof.
         ///
         /// This does not guarantee that this represents a P2TR [`Witness`]. It
         /// merely gets the second to last or third to last element depending on
-        /// the first byte of the last element being equal to 0x50.
-        ///
-        /// See [`Script::is_p2tr`] to check whether this is actually a Taproot witness.
-        fn tapscript(&self) -> Option<&Script> {
-            if self.is_empty() {
-                return None;
-            }
-
-            if self.taproot_annex().is_some() {
-                self.third_to_last().map(Script::from_bytes)
-            } else {
-                self.second_to_last().map(Script::from_bytes)
+        /// the first byte of the last element being equal to 0x50 and the associated
```

Comment:
nit `The last elements first byte is 0x50` maybe?  I don't really understand what associated version here means though.

Reply:
I think the wording was fine, no need to change it. The associated version - one from control block.